### ISSUE TITLE
Selection resets when text start scrolling due to mouse at top or bottom of the window

### ISSUE THREAD

**Issue by rgaudette** [gnunn1/tilix#2246]
When I select some text and drag the selection region past the top or bottom of the Tilix window the selecting will be reset once the text start scrolling.

**Comment by egmontkob** [comment]
Can you please give precise, step-by-step instructions to reproduce?

Description like "text start scrolling" is vague and ambiguous: Why is it scrolling? Is it because you're extending the selection? Or dragging the scrollbar? Or new input appearing in the bottom?

No matter which one I've tried out of these, I couldn't reproduce the issue.

Please give us the Tilix and VTE version numbers, too.

**Comment by rgaudette** [comment]
Sure, sorry about the vague defect report.  Here is more detailed description of how I can make this happen:
1. Ensure that there is enough content in a tilix window so that the tilix view does not contain all of the content vertically, thus there is scroll bar on the right of the tilix window.
2. Scroll the view such that there is content above the top of the view.
3. Using the mouse, start selecting (pressing the primary mouse button) anywhere in the middle of view.
4. Holding the primary mouse button down, drag the mouse cursor towards the top of the window, this highlights the text as the mouse pointer moves
5. Once the mouse pointer hits the top of the view the text in the view starts scrolling down
6. This immediately cause highlighted text to become un-highlighted
7.  Also, the text scrolling into the view is not highlighted even though the primary mouse button is still down

The expected behavior is that that highlighted text continue to extend over the text scrolling down, allowing the selection of text above the original view.

Here is the output of tilix --version
Versions
	Tilix version: 1.9.6
	VTE version: 0.76
	GTK Version: 3.24.41

Tilix Special Features
	Notifications enabled=0
	Triggers enabled=0
	Badges enabled=1

Also I am on a fully updated Ubuntu 24.04 install. Please let me know if you need any other info.

Thanks!
Rick

**Comment by egmontkob** [comment]
Thanks for the details!

Unfortuately I'm unable to reproduce the issue, the selection remains as expected. Tried on both Ubuntu 24.04 and 24.10.

Does an app perhaps continuously produce output in that terminal? VTE is known to have problems with that case. However, your report doesn't seem to be about this case.

Do you face the same problem in `gnome-terminal`?

**Comment by rgaudette** [comment]
No, this happens when the prompt is available. For example after listing a large directory and I try select more than a screen full of the listing. 

Gnome-terminal works as expected, I am able to select more than a screen full of text.

This machine has 3 monitors, configured through the gnome settings app.  I haven't made too many changes to the default gnome desktop settings.  Two that I know I have made are disabling animations with `gsettings set org.gnome.desktop.interface enable-animations false` and disabling contrast reduction with `gsettings set org.gnome.mutter.x11 reduce-contrast false`.

I am about to start a new install on a different machine, I'll install tilix early in the process to see if I can discover if/when I observe this behavior and report back.

**Comment by 237dmitry** [comment]
The issues are mostly with Gnome, I think Tilix is ​​lagging behind Gnome.

**Comment by niroy** [comment]
I'm facing the same issue on Fedora. Today I upgraded from fedora 37 to fedora 39 and then to fedora 41.
I didn't have this issue in Fedora 37, started having this issue from Fedora 39.

I'm using the Cinnamon desktop environment.
```
$ tilix --version
Versions
	Tilix version: 1.9.6
	VTE version: 0.78
	GTK Version: 3.24.43

Tilix Special Features
	Notifications enabled=0
	Triggers enabled=0
	Badges enabled=1
```
I installed `gnome-terminal` and the issue doesn't occur in that terminal, this issue is specific to tilix.
I'm attaching a video recording which show the issue, notice how the selection is lost.

https://github.com/user-attachments/assets/02c49b55-ecbf-4c2c-b6d0-696bd0cc4afc

**Comment by araujorm** [comment]
I can confirm this still happens, in Fedora with Cinnamon.

**Comment by rgaudette** [comment]
I made an observations that may be helpful in diagnosing this issue.  When org.gnome.desktop.interface text-scaling-factor is any of the values 1.0, 1.5, or 2.0 the behavior I originally described occurs.  But, when org.gnome.desktop.interface text-scaling-factor is set to 2.5 mouse selection past the window's boundaries works as expected.

**Comment by pasteluengas** [comment]
Still happening
GNOME Terminal 3.56.2 using VTE 0.80.1 +BIDI +GNUTLS +ICU +SYSTEMD
